src/tools/run-make-support/src/lib.rs

```diff
@@ -247,6 +247,34 @@ pub fn test_while_readonly<P: AsRef<Path>, F: FnOnce() + std::panic::UnwindSafe>
     success.unwrap();
 }
 
+/// Browse the directory `path` non-recursively and return the first file path which starts with `prefix` and has the
+/// file extension `extension`.
+#[track_caller]
+pub fn find_files_by_prefix_and_extension<P: AsRef<Path>>(
```

Comment:
Suggestion: *maybe* `shallow_find_files_by_prefix_and_extension` to indicate that this is not recursive?